xbmc/windowing/osx/WinSystemOSX.h

```diff
@@ -137,6 +134,7 @@ class CWinSystemOSX : public CWinSystemBase, public ITimerCallback
   XbmcThreads::EndTime<> m_dispResetTimer;
   bool m_fullscreenWillToggle;
   bool m_hasCursor{false};
+  int m_fullScreenMovingToScreen{-1};
```

Comment:
optional value might be a better choice here

Reply:
You mean std::optional? Last time I checked the sdk we're using did not support yet fully all the c++17 features. I don't remember if optional or variant. Will take a look as soon as possible.

Reply:
Both :laughing: In #23266 I have a commit that bumps the OSX target but that causes a hand full of warnings because some stuff got deprecated.

Reply:
Yes, optional. Boost also has it.

But if no access to such, then no problem.

Reply:
I'll add a todo for now and we tackle this later when we bump the SDK

Reply:
The SDK was bumped two days ago :slightly_smiling_face:

Reply:
Ahah, then I think it's time to look into the change then :)

Reply:
@neo1973 done, please take a look if possible

Reply:
looks good